File: pakistan-raid/comandos/unlock.py
import discord
from discord.ext import commands
from discord import app_commands

class UnlockCog(commands.Cog):
    def __init__(self, bot):
        self.bot = bot

    # Comando com Prefixo - Unlock
    @commands.command(name="unlock")
    @commands.has_permissions(manage_channels=True)
    async def unlock(self, ctx: commands.Context):
        """Habilita a escrita no canal (com prefixo)."""
        channel = ctx.channel
        await channel.set_permissions(ctx.guild.default_role, send_messages=True)
        await ctx.send(f"🔓 Canal `{channel.name}` desbloqueado!")

    # Comando com Slash - Unlock
    @app_commands.command(name="unlock", description="Habilita a escrita no canal")
    async def slash_unlock(self, interaction: discord.Interaction):
        """Habilita a escrita no canal (com SlashCommand)."""
        # Verifica se o usuário tem permissão para gerenciar canais
        if not interaction.user.guild_permissions.manage_channels:
            await interaction.response.send_message("Você não tem permissão para gerenciar canais!", ephemeral=True)
            return

        channel = interaction.channel
        await channel.set_permissions(interaction.guild.default_role, send_messages=True)
        await interaction.response.send_message(f"🔓 Canal `{channel.name}` desbloqueado!")

# Configura o Cog
async def setup(bot):
    await bot.add_cog(UnlockCog(bot))

File: botmod/comandos/unlock.js
const { ApplicationCommandType, PermissionFlagsBits } = require("discord.js");

module.exports = {
  name: "unlock",
  description: "[🔧 Moderação] Desbloqueie o Canal.",
  type: ApplicationCommandType.ChatInput,
  run: async (client, interaction) => {
    // Verifica se o usuário tem permissão de Gerenciar Canais
    if (!interaction.member.permissions.has(PermissionFlagsBits.ManageChannels)) {
      return interaction.reply({
        content: "Você não tem permissão de `Gerenciar Canais`.",
        ephemeral: true,
      });
    }

    // Responde ao usuário indicando que o processo de desbloqueio foi iniciado
    await interaction.reply({
      content: "Aguarde um momento... Estamos desbloqueando o canal.",
    });

    // Altera as permissões para desbloquear o canal
    await interaction.channel.permissionOverwrites.edit(interaction.guild.id, {
      SendMessages: true, // Permite que todos enviem mensagens
    });

    // Altera as permissões do usuário que usou o comando (para garantir que ele possa enviar mensagens)
    await interaction.channel.permissionOverwrites.edit(interaction.user.id, {
      SendMessages: true,
    });

    // Atualiza a resposta com sucesso
    interaction.editReply({
      content: "Canal desbloqueado com sucesso!",
    });
  },
};
